src/UBAFeeCalculator/UBAFeeSpokeCalculatorAnalog.test.ts

```diff
@@ -0,0 +1,331 @@
+            config.setRewardMultiplier(chainId, rewardMultiplier);
+            // We call the getEventFee function with the parameters we've set above.
+            const fee = getEventFee(amount, "inflow", lastRunningBalance, lastIncentiveBalance, 1, config).balancingFee;
+            // We are expecting that the fee will be positive. As a result, let's assert
+            // that the fee is greater than 0.
+            expect(fee.gt(0)).toBeTruthy();
+            // Because the fee is greater than zero, we applied no additional reward multiplier,
+            // or any other factors that would affect the fee, we expect that the fee will be
+            // equal to the integration of the balancing fee curve.
+            const integration = computePiecewiseLinearFunction(
+              config.getBalancingFeeTuples(Number(chainId)),
+              lastRunningBalance,
+              lastRunningBalance.add(amount)
+            );
+            // We expect that the fee is equal to the integration of the balancing fee curve.
+            // Note: we need to ensure that we're using the same decimals as the balancing fee curve.
+            // Note: The expectation above is that our fee is positive
+            expect(fee.toString()).toEqual(integration.toString());
+
+            // As an edge case, we can also test that if the balancing fee that is initially computed
+            // from the integral as zero, that the fee is also zero. This can be done in two ways:
+            // 1. We can set the balancingCurve to be a flat line at 0.
+            // 2. We can set the amount to be 0.
+            // We'll test both of these cases below.
+
+            // We start with zero amount
+            expect(
+              getEventFee(
+                toBNWei(0, decimalCount), // We set the amount to be 0
+                "inflow",
+                lastRunningBalance,
+                lastIncentiveBalance,
+                Number(chainId),
+                config
+              ).balancingFee.toString()
+            ).toEqual("0");
+
+            // We set the balancing fee curve to be a flat line at 0.
+            config.setBalancingFeeCurve(chainId, [[toBNWei(0, decimalCount), toBNWei(0, 0)]]);
+            expect(
+              getEventFee(
+                amount,
+                "inflow",
+                lastRunningBalance,
+                lastIncentiveBalance,
+                Number(chainId),
+                config
+              ).balancingFee.toString()
+            ).toEqual("0");
+          }
+        });
+
+        // We now want to test the specific code path where the balancing fee is negative initially
+        // but the incentive balance is not positive (i.e. 0 or negative). In this case, we expect
+        // that the fee will be zero.
+        it("should return a balancing fee of 0 if the balancing fee is initially negative and incentive balance is non-positive", () => {
```

Comment:
nit: does this test also the edge case where the balancing fee is 0 and incentive balance is < 0, should also return balancing fee of 0?

Reply:
Yes it should. We initially resolve a negative balancing fee. Then, we set `lastIncentiveBalance` to be negative and re-confirm that the balance is yielded as zero.

```
            // Finally, we can set the lastIncentiveBalance to be negative to ensure that calling
            // our fee again will result in a zero
            lastIncentiveBalance = toBNWei(Math.floor(Math.random() * -100) + 1, decimalCount);
            fee = getEventFee(amount, "outflow", lastRunningBalance, lastIncentiveBalance, 1, config).balancingFee;
            expect(fee.toString()).toEqual("0");
```